pkg/cmd/volume/rm.go

```diff
@@ -44,23 +46,28 @@ func Remove(ctx context.Context, client *containerd.Client, volumes []string, op
 
 	var volumenames []string // nolint: prealloc
 	for _, name := range volumes {
-		volume, err := volStore.Get(name, false)
-		if err != nil {
-			return err
-		}
-		if _, ok := usedVolumes[volume.Name]; ok {
+		if _, ok := usedVolumes[name]; ok {
 			return fmt.Errorf("volume %q is in use", name)
 		}
 		volumenames = append(volumenames, name)
 	}
-	removedNames, err := volStore.Remove(volumenames)
+	// if err is set, this is a hard filesystem error
+	removedNames, errs, err := volStore.Remove(volumenames)
```

Comment:
Note that on such a dramatic error (filesystem failure typically), we do not report whether and which volumes were (possibly) successfully removed.
I think it is fine - the only reason this would happen is if `os.RemoveAll` would err.